### DISCUSSION TITLE
String of Tpl

### DISCUSSION THREAD

**Discussion by shadowtime2000** [swc-project/swc#1281] [Q&A] [answered]
Sorry for the dumb question, but how do you get the basic string value of template with the Rust interface thing? Q&A was the best fitting category, there should probably a be a help category.

**Comment by kdy1** [comment, marked as answer]
You can get it by getting value of [`TplElement`](https://swc.rs/rustdoc/swc_ecma_ast/struct.Tpl.html#structfield.quasis).

**Reply by shadowtime2000** [reply]
I looked into that but it seemed `swc_ecma_ast` has a special `Str` struct which is used as the `raw` value of `TplElement` and I can't seem to figure out what `Str` is actually meant for, and how to get a string out of it.

**Reply by kdy1** [reply]
`Str` contains a field named `value`, which of type is simply an interned string.

**Reply by shadowtime2000** [reply]
I was a little confused about the Atom thing, thanks.
